### PR TITLE
GIMP-from-git snap (WIP)

### PR DESCRIPTION
**Pull request by wandrewkeech**
Add changes to `snappy-playpen/README.md` and add `gimp-git/` project directory

### CODE REVIEW

**Comment by dplanella** [comment]
Thanks so much for the contribution, you rock!

Looking at the continuous integration output, it seems that snapcraft fails. Could you add any comments about the failure and where you'd need help?

**Comment by wandrewkeech** [comment]
As dholbach as mentioned on gitter, the snap doesn't yet have glib or gtk-pixbuf as parts with deps yet.
However my biggest hangup is I don't really have a clear idea of how to issue build environment variables into the part building process, and once parts are staged how to get the later parts to see/use them as dependencies.

**Comment by wandrewkeech** [comment]
I get very consistent errors which come from not providing ACLOCAL with what it wants:

libtool: install: /usr/bin/install -c .libs/libgio-2.0.lai /root/parts/glib/install/lib/libgio-2.0.la
libtool: warning: remember to run 'libtool --finish /lib'   <-- Every component in a built part complains about not being in '/lib' when it's finished compiling

I think something like libgio is a `file` or `fileset` kind of include, where the system's shipped version is fine, and maybe needs to be included(?), but don't have a clear idea of how to do that either.

**Comment by wandrewkeech** [comment]
Holy moley, got all the way through to building gimp itself on a `snapcraft cleanbuild` just now.
Here's the error output from that: 
```
== Missing mandatory dependencies ==

GIMP configuration failed.
The following mandatory dependencies are missing:
  - gegl-0.3 >= 0.3.6
  - gegl
      *** Could not find gegl in your PATH.
  - gtk-update-icon-cache
      *** Could not find gtk-update-icon-cache in your PATH.
  - libmypaint >= 1.3.0
  - Python 2 headers
      *** Please install them, or skip building the python scripting extension by
      *** passing --disable-python to configure (but then you will not be able
      *** to use scripts for GIMP that are written in Python).

```

File: gimp-git/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+apps:
+    gimp:
+        command: gimp
+        plugs: [no-security]
+
+plugs:
+    no-security:
+        interface: old-security
+        security-template: unconfined
```
Comment (sergiusens) [inline gimp-git/snapcraft.yaml]: this does not exist anymore.

Instead use no plugs at all in the `gimp` app in `apps` and add a toplevel key which has this `confinement: devmode`

Comment (wandrewkeech) [inline gimp-git/snapcraft.yaml]: Okay, thank you.

**Comment by dplanella** [comment]
It won't solve your problem, but for the missing python 2 headers, you can probably for now do as the comment says at the end of the config step and pass `--disable-python` to the build

**Comment by dplanella** [comment]
For gegl, it seems the configure script is looking for the `$GEGL` env variable https://github.com/GNOME/gimp/blob/master/configure.ac#L655

**Comment by dplanella** [comment]
And continuing on gegl, I think the issue might be that you're pulling from master instead of the `gegl-03` branch. Or at least this seems to indicate [a similar issue](http://wiki.gimp.org/wiki/Hacking:Problems_and_solutions#GIMP_build_breaks_for_missing_dependency_gegl-0.2_.3E_0.2.0)

**Comment by wandrewkeech** [comment]
Pulling `gegl` master is a big part of `gimp-git`, as that's where nearly all the pixel buffer manipulation horsepower is, so I really want the build to include that for sure (as opposed to the documentation or i18n) while getting it to start working. 

Regarding the environment variable action, can all of those definitions just get included as 'literals' in the `configflags:` property, per part, or is there some other way? I remember seeing serguisens on an old snappy-clinic doing some things with `$SNAP` variables or something, but the examples were fast and I couldn't figure out how they were working

**Comment by dplanella** [comment]
Indeed, sorry for the noise on gegl. I think regarding build variables you can define them as `-D<var>` directly, but you might be better off asking @sergiusens or @kyrofa on gitter or #snappy on IRC.

**Comment by wandrewkeech** [comment]
Yes thank you :) 
Trying to finish some edits on the above post as progress report/notes to post into the gitter channel before I go off to do things afk this afternoon

**Comment by wandrewkeech** [comment]
Closed the request so I can keep updating the repo without blowing up the travis build history with broken crap.

**Pull request closed without merging**